### ISSUE TITLE
Tuple undefined when passed to a comptime generic parameter

### ISSUE THREAD

**Issue by Vexu** [ziglang/zig#3878]
```zig
fn Foo(comptime args: var) void {}
// test "undefined" {
//     Foo(.{1});
// }

test "not undefined" {
    comptime Foo(.{1});
}

test "not undefined" {
    Foo(1);
}

fn Bar(args: var) void {}
test "not undefined" {
    Bar(.{1});
}

fn Baz(comptime args: [1]u32) void {}
test "not undefined" {
    Baz(.{1});
}

fn Qux(args: [1]u32) void {}
test "not undefined" {
    Qux(.{1});
}
```

**Comment by alexnask** [comment]
This is fixed by https://github.com/ziglang/zig/pull/4610  
Unfortunately that PR also broke anonymous struct literal field redefinition errors.  

I will try to update it tomorrow.

**Comment by andrewrk** [comment]
Solved by @alexnask in debcc79d56a40f77b92e243b4e344fc9385bd405

**Comment by andrewrk** [comment]
I believe the fix was reverted due to causing a regression. I have a new fix for this with tests running locally.
